src/jit/lsra.cpp

```diff
@@ -2036,14 +2037,27 @@ void LinearScan::identifyCandidates()
     }
 #endif // DOUBLE_ALIGN
 
+    // initialize mapping from tracked local to interval
+    if (compiler->lvaTrackedCount > 0)
+    {
+        localVarIntervals = new (compiler, CMK_LSRA) Interval*[compiler->lvaTrackedCount];
+    }
+
     for (lclNum = 0, varDsc = compiler->lvaTable; lclNum < compiler->lvaCount; lclNum++, varDsc++)
     {
-        // Assign intervals to all the variables - this makes it easier to map
-        // them back
-        var_types intervalType = (var_types)varDsc->lvType;
-        Interval* newInt       = newInterval(intervalType);
+        Interval* newInt;
 
-        newInt->setLocalNumber(lclNum, this);
+        if (varDsc->lvTracked)
+        {
+            // Create an interval for all tracked variables, even if the tracked variable is not
+            // a register candidate. This is probably just done for consistency (e.g., code that
+            // simply walks over all tracked variables don't need to check if they are register
+            // candidates, or if the interval exists). It might be possible to only allocate
+            // intervals for register candidate tracked variables.
+
```

Comment:
+1. I think that this guess is correct:
> It might be possible to only allocate intervals for register candidate tracked variables
which is simply to say that the entries in the `localVarIntervals` table for non-candidate tracked vars would be null.